plugins/UsersManager/Model.php

```diff
@@ -245,11 +245,14 @@ public function userEmailExists($userEmail)
     public function addUserAccess($userLogin, $access, $idSites)
     {
         foreach ($idSites as $idsite) {
-            $this->getDb()->insert(Common::prefixTable("access"),
-                array("idsite" => $idsite,
-                      "login"  => $userLogin,
-                      "access" => $access)
-            );
+            $count = $this->getDb()->fetchOne("SELECT count(*) FROM " . Common::prefixTable("site"). " WHERE idsite = ?",$idsite);
+            if($count != 0){
+                 $this->getDb()->insert(Common::prefixTable("access"),
+                    array("idsite" => $idsite,
+                         "login"  => $userLogin,
+                         "access" => $access)
+                );
+            }
```

Comment:
What I meant is there should be no `"SELECT count(*) FROM " . Common::prefixTable("site"). " WHERE idsite = ?",$idsite` in this method. This method that you changed should stay unchanged. 

Instead, wherever `addUserAccess()` is called, one should ask the SitesManager API (`Piwik\API\Request::processRequest('SitesManager.getSiteFromId', array('idSite' => $idSite));`) whether the site actually exists. For example:

```php
foreach ($idSites as $idsite) {
    $site = Piwik\API\Request::processRequest('SitesManager.getSiteFromId', array('idSite' => $idSite));
    if (!empty($site)) {
         $this->model->addUserAccess($userLogin, $access, $idSite);
   }
}
```